stock_picking_invoice_link/models/stock.py

```diff
@@ -59,11 +44,34 @@ class AccountInvoice(models.Model):
 
     picking_ids = fields.One2many(
         comodel_name='stock.picking', inverse_name='invoice_id',
-        string='Related Pickings', readonly=True,
-        copy=False,
+        string='Related Pickings', readonly=True, copy=False,
         help="Related pickings "
              "(only when the invoice has been generated from the picking).")
 
+    @api.multi
+    def unlink(self):
+        if any([x and x != 'cancel' for x in
+                self.mapped('picking_ids.state')]):
+            raise exceptions.Warning(
```

Comment:
@pedrobaeza maybe this will cause a problem. Cause an invoice created from picking won't be able to be deleted if it has a picking not cancelled, even if the invoice has been cancelled.